src/ui/components/DialogSharedEntryBindings/DialogSharedEntryBindings.tsx

```diff
@@ -0,0 +1,149 @@
+import block from 'bem-cn-lite';
+import {CollectionItemEntities} from 'shared';
+import type {GetEntryResponse} from 'shared/schema';
+
+import DialogManager from '../DialogManager/DialogManager';
+import {EntitiesList} from '../EntitiesList/EntitiesList';
+import {PlaceholderIllustration} from '../PlaceholderIllustration/PlaceholderIllustration';
+import {SharedBindingsList} from '../SharedBindingsList/SharedBindingsList';
+
+import type {AttachmentValue} from './constants';
+import {Attachment, ObjectsListTitles} from './constants';
+import type {mock} from './mock';
+import {getEntityBindings} from './mock';
+
+import './DialogSharedEntryBindings.scss';
+
+type DialogSharedEntryBindingsProps = {
+    open: boolean;
+    entry: GetEntryResponse;
+    onClose: () => void;
+};
+
+export const DIALOG_SHARED_ENTRY_BINDINGS = Symbol('DIALOG_SHARED_ENTRY_BINDINGS');
+
+export interface OpenDialogSharedEntryBindingArgs {
+    id: typeof DIALOG_SHARED_ENTRY_BINDINGS;
+    props: DialogSharedEntryBindingsProps;
+}
+
+const sortEntities = (entities: (typeof mock)['items']): (typeof mock)['items'] => {
+    return entities.sort((entity) => (entity.entity === CollectionItemEntities.WORKBOOK ? -1 : 1));
+};
+
+const b = block('dialog-shared-entries-binding');
+
+export const DialogSharedEntryBindings: React.FC<DialogSharedEntryBindingsProps> = ({
+    onClose,
+    open,
+    entry,
+}) => {
+    const [currentDirection, setCurrentDirection] = React.useState<AttachmentValue>(
+        Attachment.TARGET,
+    );
+    const [entities, setEntities] = React.useState<(typeof mock)['items']>([]);
+
+    const [searchFilter, setSearchFilter] = React.useState('');
+    const [isLoading, setIsLoading] = React.useState(true);
+    const [isError, setIsError] = React.useState(false);
+
+    const showDirectionControl = entry.scope === 'dataset';
+
+    const handleDirectionChange = (value: AttachmentValue) => {
+        setCurrentDirection(value);
+    };
+
+    const fetchEntityBindings = React.useCallback(() => {
+        setIsLoading(true);
+        setIsError(false);
+        // cancelConcurrentRequest();
+        getEntityBindings(entry.entryId, currentDirection, searchFilter)
+            .then((response) => {
+                setEntities(sortEntities(response.items));
+                setIsLoading(false);
+            })
+            .catch((error) => {
+                if (error.isCancelled) {
+                    return;
+                }
+                setIsError(true);
+                setIsLoading(false);
+            });
+    }, [entry, currentDirection, searchFilter]);
+
+    React.useEffect(() => {
+        fetchEntityBindings();
+    }, [fetchEntityBindings]);
+
+    const renderRelations = () => {
+        if (isError) {
+            const renderRetryAction = () => (
+                <Button className={b('button-retry')} size="l" view="action" onClick={() => {}}>
+                    Попробовать снова
```

Comment:
Please add a comment with a text ticket for all constants. And it's better, of course, to put it all in a separate object and use it as an `i18n` mockup